api/src/main/java/com/exelaration/abstractmemery/controllers/UserController.java

```diff
@@ -0,0 +1,24 @@
+package com.exelaration.abstractmemery.controllers;
+
+import com.exelaration.abstractmemery.domains.ApplicationUser;
+import com.exelaration.abstractmemery.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+  @Autowired private UserRepository userRepository;
+  @Autowired private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+  @PostMapping("/sign-up")
+  public void signUp(@RequestBody ApplicationUser user) {
+    user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+    userRepository.save(user);
```

Comment:
This probably belongs in `UserDetailService`.  It should do the work of interfacing with the database.  Our controller should deal with the service by asking if `userExists` and if not, ask it to `saveUser`.

Reply:
I addressed this, but I wanted to ask since it kind've relates. Before the controller was using a void method, but I want it to be able to return to the front end whether or not a the user already exists so we can display that as an error. Right now I changed the controller method to return a string telling if the user exists or not. Is this fine, or should we go about this differently?

Reply:
That's fine.